integration-test-groups/cxf-soap/cxf-soap-mtom-awt/src/main/java/org/apache/camel/quarkus/component/cxf/soap/mtom/it/CxfSoapMtomRoutes.java

```diff
@@ -0,0 +1,141 @@
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+@ApplicationScoped
+public class CxfSoapMtomRoutes extends RouteBuilder {
+
+    @Inject
+    @Named("loggingMtomFeatureClient")
+    LoggingFeature loggingFeature;
+
+    @Override
+    public void configure() {
+
+        from("direct:mtomEnabledInvoker")
+                .to("cxf:bean:soapMtomEnabledClientEndpoint?dataFormat=POJO");
+
+        from("direct:mtomDisabledInvoker")
+                .to("cxf:bean:soapMtomDisabledClientEndpoint?dataFormat=POJO");
+
+        from("cxf:bean:soapMtomEnabledServerEndpoint?dataFormat=POJO")
+                .to("direct:processImage");
+
+        from("cxf:bean:soapMtomDisabledServerEndpoint?dataFormat=POJO")
+                .to("direct:processImage");
+
+        from("direct:processImage")
+                .process("imageServiceProcessor")
+                .recipientList((simple("bean:imageService?method=${header.operationName}")));
+
+    }
+
+    @ApplicationScoped
+    @Named("imageServiceProcessor")
+    static class ImageServiceProcessor implements Processor {
+        @Override
+        public void process(Exchange exchange) throws Exception {
+            String operationName = (String) exchange.getIn().getHeaders().get("operationName");
+            MessageContentsList list = (MessageContentsList) exchange.getIn().getBody();
+            if ("uploadImage".equals(operationName)) {
+                exchange.getIn().getHeaders().put("image", list.get(0));
+                exchange.getIn().getHeaders().put("imageName", list.get(1));
+                exchange.getIn().getHeaders()
+                        .put("operationName", "uploadImage(${header.image},${header.imageName})");
+            } else if ("downloadImage".equals(operationName)) {
+                exchange.getIn().setBody(list.get(0));
+            }
+        }
+
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Named("loggingMtomFeatureClient")
+    public LoggingFeature loggingFeature() {
+        final LoggingFeature result = new LoggingFeature();
+        result.setPrettyLogging(true);
+        return result;
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Named
+    CxfEndpoint soapMtomEnabledClientEndpoint() {
+        return commonCxfEndpoint(true, getServerUrl() + "/soapservice/mtom-enabled-image-service");
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Named
+    CxfEndpoint soapMtomDisabledClientEndpoint() {
+        return commonCxfEndpoint(false, getServerUrl() + "/soapservice/mtom-disabled-image-service");
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Named
+    CxfEndpoint soapMtomEnabledServerEndpoint() {
+        return commonCxfEndpoint(true, "/mtom-enabled-image-service");
+    }
+
+    @Produces
+    @ApplicationScoped
+    @Named
+    CxfEndpoint soapMtomDisabledServerEndpoint() {
+        return commonCxfEndpoint(false, "/mtom-disabled-image-service");
+    }
```

Comment:
All `@Named` names `direct:` endpoint names and WS paths should be uniquely namespaced with `MtomAwt` to avoid clashes when runnig this test together with mtom without AWS in the grouped module.